call/signaling.php

```diff
@@ -0,0 +1,208 @@
+<?php
+// --- Security Headers ---
+header('Content-Type: application/json');
+header('Access-Control-Allow-Origin: *');
+header('Access-Control-Allow-Methods: POST');
+header('Access-Control-Allow-Headers: Content-Type');
+
+// --- Helper: Sanitize input ---
+function clean($str) {
+    return preg_replace('/[^a-zA-Z0-9_\-]/', '', $str);
+}
+$room = clean($_POST['room'] ?? '');
+$user = clean($_POST['user'] ?? '');
+$type = clean($_POST['type'] ?? '');
+$target = clean($_POST['target'] ?? '');
+$data = $_POST['data'] ?? '';
+$admin_password = $_POST['admin_password'] ?? '';
+$admin_token = $_POST['admin_token'] ?? '';
+$action = clean($_POST['action'] ?? '');
+
+// --- Validate room/user ---
+if (!$room || !$user && !in_array($type, ['admin_login','get_users','get_signals','chat_get'])) {
+    http_response_code(400);
+    echo json_encode(['error' => 'Missing or invalid room/user']);
+    exit;
+}
+
+// --- Prevent directory traversal ---
+if (strpos($room, '..') !== false || strpos($user, '..') !== false) {
+    http_response_code(400);
+    echo json_encode(['error' => 'Invalid room/user']);
+    exit;
+}
+
+$dir = "rooms";
+if (!is_dir($dir)) mkdir($dir, 0700, true);
+
+$users_file = "$dir/$room-users.json";
+$signals_file = "$dir/$room-signals.txt";
+$chat_file = "$dir/$room-chat.txt";
+$admin_file = "$dir/$room-admin.json";
+
+$ROOM_ADMIN_PASSWORD = getenv('ROOM_ADMIN_PASSWORD') ?: "GoDodgers!";
```

Comment:
You never want to commit a password or secret to version control. It is not secure. Instead, it can be set as an environment variable.